micrometer-core/src/test/java/io/micrometer/core/instrument/binder/httpcomponents/PoolingNHttpClientConnectionManagerMetricsBinderTest.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2017 Pivotal Software, Inc.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micrometer.core.instrument.binder.httpcomponents;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.MockClock;
+import io.micrometer.core.instrument.simple.SimpleConfig;
+import io.micrometer.core.instrument.simple.SimpleMeterRegistry;
+import org.apache.http.impl.nio.conn.PoolingNHttpClientConnectionManager;
+import org.apache.http.pool.PoolStats;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * Unit tests for {@link PoolingNHttpClientConnectionManagerMetricsBinder}.
+ *
+ * @author Daniel Albuquerque
+ */
+class PoolingNHttpClientConnectionManagerMetricsBinderTest {
+
+    private MeterRegistry registry = new SimpleMeterRegistry(SimpleConfig.DEFAULT, new MockClock());
+    private PoolingNHttpClientConnectionManager connectionManager;
+    private PoolingNHttpClientConnectionManagerMetricsBinder binder;
+
+    @BeforeEach
+    void setup() {
+        connectionManager = mock(PoolingNHttpClientConnectionManager.class);
```

Comment:
Would there be any issues instantiating a connection pool for the test? It seems like that would be better than mocking. Though I suppose we would need a wiremock server to make HTTP requests with the HTTP client, and that would add some complexity.

Reply:
I would love to add testcontainers and get rid of some of these mocks :/